CMake/OtherTests.cmake

```diff
@@ -177,20 +177,48 @@ int main(void) {
   return 0;
 }" HAVE_STRUCT_TIMEVAL)
 
-
-include(CheckCSourceRuns)
-# See HAVE_POLL in CMakeLists.txt for why poll is disabled on macOS
-if(NOT APPLE)
-  set(CMAKE_REQUIRED_FLAGS)
+if(NOT DEFINED CMAKE_TOOLCHAIN_FILE)
```

Comment:
I don't think it's right. There are other ways of defining the toolchain file (for example dockcross uses Env variable). Also not every toolchain file is in fact cross-compiling.
This: https://gitlab.kitware.com/cmake/community/wikis/doc/cmake/CrossCompiling#system-introspection is probably the way to go.

Reply:
@jzakrzewski I'm trying to support recommended way for cross-compilation in CMake. 
For curl we have #3268 which using `CMAKE_TOOLCHAIN_FILE` and this will fix the issue.
I'm not trying to support dockcross so far.

If you already have cmake snippet for cross detectin - could you please share?
I've checked your proposal. It seems too complicated for me. 
Need more time to investigate it though.

Reply:
@jzakrzewski Looks like dockcross anyway use CMAKE_TOOLCHAIN_FILE. See here https://github.com/dockcross/dockcross/tree/master/linux-arm64 
They reconfigure it with some variables, but anyway use it during cross-compilation.

Reply:
Damn, I've missed that script.
In any case, it seems (as far as I can understand the link I've pasted) that the right way is to check CMAKE_CROSSCOMPILING. I use Toolchain files to choose, well, the right toolchain - I have few compilers and, instead of fiddling with separate variables, I have a collection of toolchain files. None of them does actual cross compilation.

I don't suggest that we use full power of the `try_run` (mainly because I'm yet to learn how it works) but it seems `CMAKE_CROSSCOMPILING` is the variable to test (see also https://cmake.org/cmake/help/v3.0/variable/CMAKE_CROSSCOMPILING.html ). I may be wrong though as I have tried a real cross-compiling only a few times (with mixed results ;) )
It may also be that this check is so unimportant that it doesn't really matter if it's switched off too eagerly, in which case just merge it ;)

Reply:
@jzakrzewski According to [the latest documentation](https://cmake.org/cmake/help/latest/variable/CMAKE_CROSSCOMPILING.html)

> the use of CMAKE_CROSSCOMPILING is not recommended for projects targeting Apple devices.

Reply:
> In any case, it seems (as far as I can understand the link I've pasted) that the right way is to check CMAKE_CROSSCOMPILING. I use Toolchain files to choose, well, the right toolchain - I have few compilers and, instead of fiddling with separate variables, I have a collection of toolchain files. None of them does actual cross compilation.
> 

Perhaps you're using this wrong :) 
And in you case you'll force to `select` instead of `poll`. But it should not be critical for curl :)